### PR TITLE
Merge ek/glow to master

### PR DESCRIPTION
**Pull request by eugeneko**

### CODE REVIEW

File: Source/Samples/106_BakedLighting/BakedLighting.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
+//
+// Copyright (c) 2020-2020 the rbfx project.
```
Comment (rokups) [inline Source/Samples/106_BakedLighting/BakedLighting.cpp]: Ought to be `Copyright (c) 2017-2020 the rbfx project.` 3 years already wow.

File: Source/Samples/106_BakedLighting/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+if (NOT URHO3D_PHYSICS)
```
Comment (rokups) [inline Source/Samples/106_BakedLighting/CMakeLists.txt]: Sample does not (or at least should not) depend on physics.

File: Source/ThirdParty/CMakeLists.txt [hunk 3]
```diff
@@ -44,6 +44,30 @@ add_subdirectory(ETCPACK)
...
+        set (EMBREE_RAY_MASK ON CACHE BOOL "")
+        set (EMBREE_ISA_SSE2 ON CACHE BOOL "")
+        set (EMBREE_ISA_SSE42 OFF CACHE BOOL "")
+        set (EMBREE_ISA_AVX OFF CACHE BOOL "")
+        set (EMBREE_ISA_AVX2 OFF CACHE BOOL "")
+        set (EMBREE_ISA_AVX512KNL OFF CACHE BOOL "")
+        set (EMBREE_ISA_AVX512SKX OFF CACHE BOOL "")
+
+        # TODO: Try to remove CMAKE_INCLUDE_CURRENT_DIR at all
+        set (CMAKE_INCLUDE_CURRENT_DIR OFF)
```
Comment (rokups) [inline Source/ThirdParty/CMakeLists.txt]: This is already removed from root cmake file so you can get rid of this workaround.

File: Source/Urho3D/Glow/LightmapCharter.cpp [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+    // Try existing charts
+    unsigned chartIndex = 0;
+    for (LightmapChart& lightmapDesc : charts)
+    {
+        IntVector2 paddedPosition;
+        if (lightmapDesc.allocator_.Allocate(paddedSize.x_, paddedSize.y_, paddedPosition.x_, paddedPosition.y_))
+        {
+            const IntVector2 position = paddedPosition + padding * IntVector2::ONE;
+            return { chartIndex, position, size, settings.lightmapSize_ };
```
Comment (rokups) [inline Source/Urho3D/Glow/LightmapCharter.cpp]: I thought we dont do spaces inside curlies

File: Source/Urho3D/Graphics/LightBakingSettings.h [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    /// Global illumination data file.
+    ea::string giDataFileName_{ "GI.bin" };
+    /// Lightmap name prefix.
+    ea::string lightmapNamePrefix_{ "Textures/Lightmap-" };
+    /// Lightmap name suffix.
+    ea::string lightmapNameSuffix_{ ".png" };
+    /// Light probe group name prefix.
+    ea::string lightProbeGroupNamePrefix_{ "Binary/LightProbeGroup-" };
+    /// Light probe group name suffix.
+    ea::string lightProbeGroupNameSuffix_{ ".bin" };
```
Comment (rokups) [inline Source/Urho3D/Graphics/LightBakingSettings.h]: Prefix and suffix strings could be replaced with a single string that defines format. Like `ea::string lightProbeGroupNameFormat_{"Binary/LightProbeGroup-{}.bin"};` which you can use as `Format(lightProbeGroupNameFormat_, ...)` later.

File: Source/Urho3D/IO/File.h [hunk 6]
```diff
@@ -137,6 +140,8 @@ class URHO3D_API File : public Object, public AbstractFile
 
     /// File name.
     ea::string fileName_;
+    /// Native file name.
+    ea::string nativeFileName_;
```
Comment (rokups) [inline Source/Urho3D/IO/File.h]: Is there really a need for this? We have `GetNativePath()` in `FileSystem.h`.

File: Source/Urho3D/Math/TetrahedralMesh.cpp [hunk 7]
```diff
@@ -0,0 +1,1251 @@
...
+        if (t < -1)
+            t = -1;
+        if (t > 1)
+            t = 1;
+        t = std::acos(t);
+        a /= 3;
+        q = -2 * std::sqrt(q);
+        result[0] = q * std::cos(t / 3) - a;
+        result[1] = q * std::cos((t + M_PI * 2) / 3) - a;
+        result[2] = q * std::cos((t - M_PI * 2) / 3) - a;
```
Comment (rokups) [inline Source/Urho3D/Math/TetrahedralMesh.cpp]: Any particular reason for using these from stdlib and not from `MathDefs.h`?

Comment (eugeneko) [reply Source/Urho3D/Math/TetrahedralMesh.cpp]: I don't want to slow down performance-critical path of code by doing redundant math ops for sake of consistency. Will add comment.

File: Source/Urho3D/IO/File.h (hunk 6 above)
Comment (eugeneko) [reply Source/Urho3D/IO/File.h]: The problem is that `fileName_` _is not actually a file name_.
It is relative to resource cache directories and therefore worthless for my needs. Will add comment.

**Pull request merged**